src/golang/lib/workflow/operator/base.go

```diff
@@ -323,6 +323,11 @@ func (bfo *baseFunctionOperator) jobSpec(
 		}
 	}
 
+	expectedOutputTypes := make([]string, 0, 1)
+	for _, output := range bfo.outputs {
+		expectedOutputTypes = append(expectedOutputTypes, string(output.Type()))
+	}
```

Comment:
I kept the output types as a list since I want to try prototyping multiple-outputs soon, and this points us in the right direction for that.